core/domain/subscription_services.py

```diff
@@ -40,15 +40,32 @@ def subscribe_to_thread(user_id, feedback_thread_id):
         user_id: str. The user ID of the new subscriber.
         feedback_thread_id: str. The ID of the feedback thread.
     """
+    subscribe_to_threads(user_id, [feedback_thread_id])
+
+
+def subscribe_to_threads(user_id, feedback_thread_ids):
+    """Subscribes a user to feedback threads.
+
+    WARNING: Callers of this function should ensure that the user_id and
+    the feedback_thread_ids are valid.
+
+    Args:
+        user_id: str. The user ID of the new subscriber.
+        feedback_thread_ids: list(str). The IDs of the feedback threads.
+    """
     subscriptions_model = user_models.UserSubscriptionsModel.get(
         user_id, strict=False)
     if not subscriptions_model:
         subscriptions_model = user_models.UserSubscriptionsModel(id=user_id)
 
-    if (feedback_thread_id not in
-            subscriptions_model.general_feedback_thread_ids):
-        subscriptions_model.general_feedback_thread_ids.append(
-            feedback_thread_id)
+    # Use a set for more efficient look up.
+    current_feedback_thread_ids_set = set(
+        subscriptions_model.general_feedback_thread_ids
+    )
+    for feedback_thread_id in feedback_thread_ids:
```

Comment:
You could probably just do a set difference? I.e. list(set(feedback_thread_ids).difference(current_feedback_thread_ids_set))?